Mathlib/NumberTheory/LSeries/HurwitzZetaOdd.lean

```diff
@@ -0,0 +1,592 @@
+    (by ring : ↑π * I * ↑a ^ 2 * (I * ↑x) = I ^ 2 * ↑π * ↑a ^ 2 * x), I_sq, neg_one_mul]
+
+lemma oddKernel_undef (a : UnitAddCircle) {x : ℝ} (hx : x ≤ 0) : oddKernel a x = 0 := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  rw [← ofReal_eq_zero, oddKernel_def', jacobiTheta₂_undef, jacobiTheta₂'_undef, zero_div, zero_add,
+    mul_zero, mul_zero] <;>
+  rwa [I_mul_im, ofReal_re]
+
+/-- Auxiliary function appearing in the functional equation for the odd Hurwitz zeta kernel, equal
+to `∑ (n : ℕ), 2 * n * sin (2 * π * n * a) * exp (-π * n ^ 2 * x)`. See `hasSum_nat_sinKernel`
+for the defining sum. -/
+@[irreducible] def sinKernel (a : UnitAddCircle) (x : ℝ) : ℝ :=
+  (show Function.Periodic (fun ξ : ℝ ↦ re (jacobiTheta₂' ξ (I * x) / (-2 * π))) 1 by
+    intro ξ; simp_rw [ofReal_add, ofReal_one, jacobiTheta₂'_add_left]).lift a
+
+lemma sinKernel_def (a x : ℝ) : ↑(sinKernel ↑a x) = jacobiTheta₂' a (I * x) / (-2 * π) := by
+  rw [sinKernel, Function.Periodic.lift_coe, re_eq_add_conj, map_div₀, jacobiTheta₂'_conj]
+  simp_rw [map_mul, conj_I, conj_ofReal, map_neg, map_ofNat, neg_mul, neg_neg]
+  ring
+
+lemma sinKernel_undef (a : UnitAddCircle) {x : ℝ} (hx : x ≤ 0) : sinKernel a x = 0 := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  rw [← ofReal_eq_zero, sinKernel_def, jacobiTheta₂'_undef, zero_div]
+  rwa [I_mul_im, ofReal_re]
+
+lemma oddKernel_neg (a : UnitAddCircle) (x : ℝ) : oddKernel (-a) x = -oddKernel a x := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  rw [← ofReal_inj, ← QuotientAddGroup.mk_neg, oddKernel_def, ofReal_neg, ofReal_neg, oddKernel_def,
+    jacobiTheta₂''_neg_left]
+
+lemma oddKernel_zero (x : ℝ) : oddKernel 0 x = 0 := by
+  simpa only [neg_zero, eq_neg_self_iff] using oddKernel_neg 0 x
+
+lemma sinKernel_neg (a : UnitAddCircle) (x : ℝ) :
+    sinKernel (-a) x = -sinKernel a x := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  rw [← ofReal_inj, ← QuotientAddGroup.mk_neg, ofReal_neg, sinKernel_def, sinKernel_def, ofReal_neg,
+    jacobiTheta₂'_neg_left, neg_div]
+
+lemma sinKernel_zero (x : ℝ) : sinKernel 0 x = 0 := by
+  simpa only [neg_zero, eq_neg_self_iff] using sinKernel_neg 0 x
+
+/-- The odd kernel is continuous on `Ioi 0`. -/
+lemma continuousOn_oddKernel (a : UnitAddCircle) : ContinuousOn (oddKernel a) (Ioi 0) := by
+  induction' a using QuotientAddGroup.induction_on' with a
+  suffices ContinuousOn (fun x ↦ (oddKernel a x : ℂ)) (Ioi 0) from
+    (continuous_re.comp_continuousOn this).congr fun a _ ↦ (ofReal_re _).symm
+  simp_rw [oddKernel_def' a]
+  refine fun x hx ↦ ((Continuous.continuousAt ?_).mul ?_).continuousWithinAt
+  · exact continuous_exp.comp (continuous_const.mul continuous_ofReal)
+  · have hx' : 0 < im (I * x) := by rwa [I_mul_im, ofReal_re]
+    have hf : Continuous fun u : ℝ ↦ (a * I * u, I * u) := by
+      rw [continuous_prod_mk]
+      constructor <;> exact continuous_const.mul continuous_ofReal
```

Comment:
```suggestion
  · fun_prop
  · have hx' : 0 < im (I * x) := by rwa [I_mul_im, ofReal_re]
    have hf : Continuous fun u : ℝ ↦ (a * I * u, I * u) := by fun_prop
```
`fun_prop` is actually faster here!

It might even be possible to mark the continuity statements for Jacobi theta functions so that `fun_prop` can use them; then most of the proff could just be `fun_prop`.